src/Middleware/Session/src/DistributedSession.cs

```diff
@@ -346,6 +348,11 @@ private void Serialize(Stream output)
         {
             output.WriteByte(SerializationRevision);
             SerializeNumAs3Bytes(output, _store.Count);
+            if (!IsAvailable)
```

Comment:
This is a weird place for this check (and a regression?). You added this here because IdBytes isn't initialized? It'd be better to skip calling Serialize or Commit at all if we don't think it can succeed.

Reply:
Previously this would have resulted in a null-ref. I can update the code to no-op Commit.